src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -308,6 +308,9 @@ def add_network_rule(cmd, client, resource_group_name, account_name, action='All
         VirtualNetworkRule = cmd.get_models('VirtualNetworkRule')
         if not rules.virtual_network_rules:
             rules.virtual_network_rules = []
+        vnet_rules = [r for r in rules.virtual_network_rules if r.virtual_network_resource_id.lower() == subnet.lower()]
+        for vnet_rule in vnet_rules:
+            rules.virtual_network_rules.remove(vnet_rule)
```

Comment:
`rules.virtual_network_rules = [r for r in rules.virtual_network_rules if r.virtual_network_resource_id.lower() != subnet.lower()]`    It might be more concise?

Reply:
Yes, good catch.